python/lsst/cell_coadds/_cellCoaddBuilder.py

```diff
@@ -0,0 +1,385 @@
+            An iterable of `lsst.daf.butler.DeferredDatasetHandle` objects,
+            where the objects can be either calexp or warp images.
+        skyInfo: `pipeBase.Struct`
+            Struct with geommetric information about the patches and cells.
+        quantumDataId: `DataCoordinate`
+            An immutable dataID dictionary that uniquely refers to the dataset.
+
+        Returns
+        -------
+        multipleCellCoadd: `MultipleCellCoadd`
+            Cell-based coadded image.
+        """
+
+        cellCoadds: list[SingleCellCoadd] = []
+        patchInfo: PatchInfo = skyInfo.patchInfo  # type: ignore[attr-defined]
+        common = CommonComponents(
+            units=CoaddUnits.nJy,
+            wcs=patchInfo.wcs,
+            band=quantumDataId.get("band", None),
+            identifiers=PatchIdentifiers.from_data_id(quantumDataId),
+        )
+
+        # This for loop could/should be parallelized, may be with asyncio?
+        for cellInfo in patchInfo:
+            # Select calexps that completely overlap with the cell
+            try:
+                bbox_list = self._select_overlaps(expList, cellInfo=cellInfo, skyInfo=skyInfo)
+            except KeyError:
+                continue  # Exception handling is likely a relic, should be removed.
+
+            if not bbox_list:
+                continue  # We should raise exception here, since the task would fail anyway eventually.
+
+            scc_inputs = {
+                ObservationIdentifiers.from_data_id(handle.ref.dataId): (handle, bbox)
+                for handle, bbox in zip(expList, bbox_list)
+            }
+            if len(scc_inputs) == 0:
+                continue
+
+            result = self.singleCellCoaddBuilder.run(scc_inputs, cellInfo)
+            identifiers = CellIdentifiers(
+                cell=GridIdentifiers.from_info(cellInfo),
+                skymap=common.identifiers.skymap,
+                tract=common.identifiers.tract,
+                patch=common.identifiers.patch,
+                band=common.identifiers.band,
+            )
+            # TODO: singleCellCoaddBuilder.run should return a SingleCellCoadd
+            cellCoadd = SingleCellCoadd(
+                outer=result.image_planes,  # type: ignore[attr-defined]
+                psf=result.psf,  # type: ignore[attr-defined]
+                inner_bbox=cellInfo.inner_bbox,
+                inputs=result.inputs,  # type: ignore[attr-defined]
+                common=common,
+                identifiers=identifiers,
+            )
+            cellCoadds.append(cellCoadd)
+
+        # grid has no notion about border or inner/outer boundaries.
+        # So we have to clip the outermost border when constructing the grid.
+        grid_bbox = patchInfo.outer_bbox.erodedBy(patchInfo.getCellBorder())
+        grid = UniformGrid(grid_bbox, patchInfo.getCellInnerDimensions())
+
+        _mCellCoadd = MultipleCellCoadd(
```

Comment:
Why leading-underscore here?

Reply:
I've removed the leading underscore now. That was just a mental note to myself to remove it and just have `return MultipleCellCoadd(...` as I had it in earlier version of the code. I guess tail call optimization is not a big thing here, so it should be okay to capture the coadd with a variable and return it. It certainly made debugging easier.